trusted-firmware-m/platform/ext/target/nordic_nrf/common/nrf9160/partition/flash_layout.h

```diff
@@ -149,6 +151,11 @@
                                          FLASH_ITS_AREA_SIZE)
 #define FLASH_NV_COUNTERS_AREA_SIZE     (FLASH_AREA_IMAGE_SECTOR_SIZE)
 
+/* Non-secure storage region */
+#define NRF_FLASH_NS_STORAGE_AREA_OFFSET    (FLASH_TOTAL_SIZE - \
```

Comment:
This should be calculated from `(FLASH_NV_COUNTERS_AREA_OFFSET + FLASH_NV_COUNTERS_AREA_SIZE)` to avoid overlapping regions. Also, add a check after, to see that the regions don't go over `FLASH_TOTAL_SIZE`.

Reply:
Storage partition is always at the end so I would prefer leave it as it is and simply add a check that NV Counters and NS Storage don't overlap. Would that be fine?

Reply:
That works. As long as we prevent overlapping I don't have a strong preference.